### PR TITLE
refactor: monorepo alloy

### PR DESCRIPTION
**Pull request by KaviiSuri**
This PR introduces a monorepo structure as discussed on the last community call. it also adds all the github actions required to build test and lint things.

### CODE REVIEW

File: apps/playground-web/components/CLI/CLI.tsx
Comment (lucifercr07) [inline apps/playground-web/components/CLI/CLI.tsx:1]: Shall we rename the file to `Shell.tsx` as Arpit recommended?

File: .vscode/settings.json
Comment (lucifercr07) [inline .vscode/settings.json:1]: Is this required to be checked in?

File: apps/playground-web/components/Shell/hooks/useShell.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ export const useShell = (decreaseCommandsLeft: () => void) => {
   const inputRef = useRef<HTMLInputElement>(null);
 
   const handleCommandWrapper = () => {
-    const commandName = command.trim().split(' ')[0].toUpperCase(); // Extract the command
+    const commandName = command.trim().split(' ')[0]!.toUpperCase(); // Extract the command
```
Comment (lucifercr07) [inline apps/playground-web/components/Shell/hooks/useShell.tsx:22]: Shall we have safe check here?

**Review by lucifercr07 — APPROVED** [review]
LGTM, minor comments

File: .vscode/settings.json
Comment (KaviiSuri) [reply .vscode/settings.json:1]: imo yes, we can add recommended plugins here and usrs would have them by default.

File: apps/playground-web/components/CLI/CLI.tsx
Comment (KaviiSuri) [reply apps/playground-web/components/CLI/CLI.tsx:1]: renamed

File: apps/playground-web/components/Shell/hooks/useShell.tsx (hunk 1 above)
Comment (KaviiSuri) [reply apps/playground-web/components/Shell/hooks/useShell.tsx]: added

**Pull request merged**